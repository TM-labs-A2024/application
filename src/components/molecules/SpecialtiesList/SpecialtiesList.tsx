import { List, Stack, Heading } from '@chakra-ui/react'
import Specialty from '@src/components/atoms/Cards/Specialty'
import { Patient } from '@src/types'
import React from 'react'

export default function SpecialtiesList({
  specialties,
  label,
  patient
}: {
  patient?: Patient
  specialties: { name: string; id: string }[]
  label: string
}) {
  return (
    <Stack spacing={3} className="h-full" data-testid="specialties-list">
      {specialties.length > 0 && label && (
        <Heading as="h2" size="sm" noOfLines={1} className="mb-4">
          {label}
        </Heading>
      )}
      <List spacing={3} className="h-full overflow-scroll pb-24">
        {specialties.map((specialty) => (
          <Specialty
            name={specialty.name}
            id={String(specialty.id)}
            key={`specialty-${specialty.id}`}
            patient={patient}
          />
        ))}
      </List>
    </Stack>
  )
}
