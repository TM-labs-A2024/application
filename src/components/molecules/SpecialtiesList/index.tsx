import SpecialtiesList from './SpecialtiesList'

export default SpecialtiesList
