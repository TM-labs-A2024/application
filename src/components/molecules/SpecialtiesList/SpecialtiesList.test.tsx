import { specialties } from '@src/constants'
import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import SpecialtiesList from './SpecialtiesList'

const props = {
  specialties,
  label: 'label'
}

describe('Molecules > SpecialtiesList test', () => {
  test('The component shows the label if the specialties list has at least one entry', () => {
    render(<SpecialtiesList {...props} />)

    expect(screen.getByText(props.label)).toBeInTheDocument()
  })

  test('The quantity of children is the expected', () => {
    render(<SpecialtiesList {...props} />)

    expect(screen.getAllByTestId('specialty-card')).toHaveLength(48)
  })

  test('Matches the snapshot', () => {
    render(<SpecialtiesList {...props} />)

    expect(screen.getByTestId('specialties-list')).toMatchSnapshot()
  })
})
