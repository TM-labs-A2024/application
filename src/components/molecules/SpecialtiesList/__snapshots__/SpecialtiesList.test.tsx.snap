// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules > SpecialtiesList test Matches the snapshot 1`] = `
<div
  class="chakra-stack h-full css-11nrrcx"
  data-testid="specialties-list"
>
  <h2
    class="chakra-heading mb-4 css-o8iskg"
  >
    label
  </h2>
  <ul
    class="h-full overflow-scroll pb-24 css-1xa84ef"
    role="list"
  >
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/1"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1mw972v"
          >
            <div
              aria-label="Alergología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              A
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Alergología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/2"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1suxtbl"
          >
            <div
              aria-label="Anestesiología y reanimación"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ar
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Anestesiología y reanimación
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/3"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-11uplnc"
          >
            <div
              aria-label="Aparato digestivo"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ad
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Aparato digestivo
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/4"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-d35ey5"
          >
            <div
              aria-label="Cardiología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              C
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cardiología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/5"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-zofkk0"
          >
            <div
              aria-label="Endocrinología y nutrición"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              En
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Endocrinología y nutrición
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/6"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1vg0ixc"
          >
            <div
              aria-label="Geriatría"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              G
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Geriatría
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/7"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-15s93ye"
          >
            <div
              aria-label="Hematología y hemoterapia"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Hh
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Hematología y hemoterapia
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/8"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1mhmnk4"
          >
            <div
              aria-label="Medicina de la educación física y del deporte"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Md
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina de la educación física y del deporte
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/9"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1oyq4fl"
          >
            <div
              aria-label="Medicina espacial"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Me
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina espacial
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/10"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-s9b35e"
          >
            <div
              aria-label="Medicina intensiva"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mi
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina intensiva
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/11"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1lzewqe"
          >
            <div
              aria-label="Medicina interna"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mi
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina interna
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/12"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-u80mju"
          >
            <div
              aria-label="Medicina legal y forense"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mf
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina legal y forense
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/13"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1ngu9s0"
          >
            <div
              aria-label="Medicina preventiva y salud pública"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mp
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina preventiva y salud pública
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/14"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1u3wgk6"
          >
            <div
              aria-label="Medicina del trabajo"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mt
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina del trabajo
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/15"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-gimo61"
          >
            <div
              aria-label="Nefrología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              N
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Nefrología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/16"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-wjjp38"
          >
            <div
              aria-label="Neumología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              N
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Neumología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/17"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1ee3np8"
          >
            <div
              aria-label="Neurología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              N
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Neurología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/18"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-qkyhbb"
          >
            <div
              aria-label="Neurofisiología Clínica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              NC
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Neurofisiología Clínica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/19"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-qgl2wl"
          >
            <div
              aria-label="Oncología médica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Om
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Oncología médica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/20"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-5lwjjp"
          >
            <div
              aria-label="Oncología radioterápica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Or
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Oncología radioterápica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/21"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-l1zexa"
          >
            <div
              aria-label="Pediatría"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              P
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Pediatría
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/22"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-dbvfq3"
          >
            <div
              aria-label="Psiquiatría"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              P
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Psiquiatría
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/23"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-ybpr2k"
          >
            <div
              aria-label="Rehabilitación"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              R
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Rehabilitación
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/24"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1zokjr"
          >
            <div
              aria-label="Reumatología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              R
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Reumatología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/25"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1x3rnax"
          >
            <div
              aria-label="Medicina familiar y comunitaria"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mc
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina familiar y comunitaria
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/26"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1ftrvw7"
          >
            <div
              aria-label="Cirugía cardiovascular"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Cc
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía cardiovascular
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/27"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-2ecao8"
          >
            <div
              aria-label="Cirugía general y del aparato digestivo"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Cd
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía general y del aparato digestivo
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/28"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-16ekup4"
          >
            <div
              aria-label="Cirugía oral y maxilofacial"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Cm
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía oral y maxilofacial
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/29"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1gemmtm"
          >
            <div
              aria-label="Cirugía ortopédica y traumatología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ct
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía ortopédica y traumatología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/30"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-7ev6l9"
          >
            <div
              aria-label="Cirugía pediátrica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Cp
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía pediátrica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/31"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-boq4vy"
          >
            <div
              aria-label="Cirugía plástica, estética y reparadora"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Cr
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía plástica, estética y reparadora
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/32"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1roymqh"
          >
            <div
              aria-label="Cirugía torácica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ct
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Cirugía torácica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/33"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-kyhb3k"
          >
            <div
              aria-label="Neurocirugía"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              N
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Neurocirugía
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/34"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1tvyueo"
          >
            <div
              aria-label="Angiología y cirugía vascular"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Av
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Angiología y cirugía vascular
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/35"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-qsxtja"
          >
            <div
              aria-label="Dermatología médico-quirúrgica y venereología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Dv
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Dermatología médico-quirúrgica y venereología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/36"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1ubjzeo"
          >
            <div
              aria-label="Obstetricia y ginecología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Og
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Obstetricia y ginecología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/37"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1a8iloz"
          >
            <div
              aria-label="Oftalmología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              O
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Oftalmología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/38"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1l74pql"
          >
            <div
              aria-label="Otorrinolaringología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              O
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Otorrinolaringología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/39"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-kt881q"
          >
            <div
              aria-label="Urología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              U
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Urología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/40"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-biv2cz"
          >
            <div
              aria-label="Análisis clínicos"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ac
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Análisis clínicos
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/41"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-wp2wfn"
          >
            <div
              aria-label="Anatomía patológica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Ap
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Anatomía patológica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/42"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1cpoeex"
          >
            <div
              aria-label="Bioquímica clínica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Bc
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Bioquímica clínica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/43"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-17z445n"
          >
            <div
              aria-label="Farmacología clínica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Fc
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Farmacología clínica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/44"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-ba2oa2"
          >
            <div
              aria-label="Inmunología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              I
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Inmunología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/45"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-zwxkat"
          >
            <div
              aria-label="Medicina nuclear"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mn
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Medicina nuclear
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/46"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-1hd0yeh"
          >
            <div
              aria-label="Microbiología y parasitología"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Mp
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Microbiología y parasitología
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/47"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-hxkvh2"
          >
            <div
              aria-label="Neurofisiología clínica"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              Nc
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Neurofisiología clínica
          </p>
        </div>
        <div />
      </div>
    </a>
    <a
      class="chakra-link border-1 flex w-full rounded-md border border-gray-400 p-4 css-spn4bz"
      data-testid="specialty-card"
      href="/especialidad/48"
    >
      <div
        class="flex w-full flex-row items-center justify-between"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="chakra-avatar css-ycb1az"
          >
            <div
              aria-label="Radiodiagnóstico"
              class="chakra-avatar__initials css-1ebyn6"
              role="img"
            >
              R
            </div>
          </span>
          <p
            class="chakra-text ml-4 css-0"
          >
            Radiodiagnóstico
          </p>
        </div>
        <div />
      </div>
    </a>
  </ul>
</div>
`
